File: MBApplicationUI/src/app/model/account.ts
import { Currency } from "./currency";

export class Account {
  public id: number;
  public number: number;
  public defaultCurrencyId: number;
  public defaultCurrency: Currency;
  public userId: number;

  constructor(id: number, number: number, defaultCurrencyId: number, userId: number){
    this.id = id;
    this.number = number;
    this.defaultCurrencyId = defaultCurrencyId;
    this.userId = userId;
    this.defaultCurrency = new Currency(0,'','');
  }
}

File: MB.Data.Entities/Account.cs
using System.Collections.Generic;

namespace MB.Data.Entities
{
    public class Account
    {
        public int ID { get; set; }
        public int Number { get; set; }

        public int DefaultCurrencyId { get; set; }
        public virtual Currency DefaultCurrency { get; set; }

        public int UserId { get; set; }
        public virtual User User { get; set; }

        public virtual ICollection<Transaction> Transactions { get; set; }
    }
}
